corgidrp/walker.py

```diff
@@ -0,0 +1,128 @@
+    "update_to_l2a" : corgidrp.l1_to_l2a.update_to_l2a
+}
+
+recipe_dir = os.path.join(os.path.dirname(__file__), "recipe_templates")
+
+def autogen_reicpe(filelist, outputdir, template=None):
+    """
+    Automatically creates a recipe by identifyng and populating a template
+
+    Args:
+        filelist (list of str): list of filepaths to files
+        outputdir (str): output directory folderpath
+
+    """
+    # load the first frame to check what kind of data and identify recipe
+    first_frame = data.autoload(filelist[0])
+    
+    recipe_filename = guess_template(first_frame)
+
+    # load the template recipe
+    recipe_filepath = os.path.join(recipe_dir, recipe_filename)
+    template = json.load(open(recipe_filepath, 'r'))
+
+    # create the personalized recipe
+    recipe = template.copy()
+    recipe["template"] = False
+
+    for filename in filelist:
+        recipe["inputs"].append(filename)
+    
+    recipe["outputdir"] = outputdir
+
+    ## Populate calibration files that need to be automatically populated
+    this_caldb = caldb.CalDB()
+    for step in recipe["steps"]:
+        if "calibs" in step:
+            for calib in step["calibs"]:
+                # order matters, so only one calibration file per dictionary
+                
+                if step["calibs"][calib].upper() == "AUTOMATIC":
+                    calib_dtype = data.datatypes[calib]
+                    best_cal_file = this_caldb.get_calib(first_frame, calib_dtype)
+                    # set calibration file to this one
+                    step["calibs"][calib] = best_cal_file.filepath
+
+    return recipe
+
+
+def guess_template(image):
+    """
+    Guesses what template should be used to process a specific image
+
+    Args:
+        image (corgidrp.data.Image): an Image file to process
+
+    Returns
+        str: the best template filename
+    """
+    if image.ext_hdr['DATA_LEVEL'] == "L1":
+        if image.pri_hdr['OBSTYPE'] == "ENG":
+            recipe_filename = "l1_to_l2a_eng.json"
+        else:
+            recipe_filename = "l1_to_l2a_basic.json"
+    else:
+        raise NotImplementedError()
+
+    return recipe_filename
+
+def run_recipe(recipe):
+    """
+    Run the specified recipe
+
+    Args:
+        recipe (dict or str): either the filepath to the recipe or the already loaded in recipe
+    """
+    if isinstance(recipe, str):
+        # need to load in
+        recipe = json.load(open(recipe, "r"))
+    
+    # configure pipeline as needed
+    for setting in recipe['drpconfig']:
+        # equivalent to corgidrp.setting = recipe['drpconfig'][setting]
+        setattr(corgidrp, setting, recipe['drpconfig'][setting])
+
+    # read in data
+    filelist = recipe["inputs"]
+    curr_dataset = data.Dataset(filelist)
+    
+    # execute each pipeline step
+    for step in recipe["steps"]:
+        if step["name"].lower() == "save":
+            # special save instruction
```

Comment:
After this PR I think we'll want to add a feature request to enable adding suffixes easily for a "save" recipe call.

Reply:
good idea!